### DISCUSSION TITLE
Point2Grid WARNING: regrid_goes_variable() -> No valid data!

### DISCUSSION THREAD

**Discussion by Ho-ChunHuang-NOAA** [dtcenter/METplus#2738] [Statistical Computation] [answered]
The GOES ABI pixel AOD will only have values during daytime hours (or if there is a sunlight).  During the nighttime or hour around dawn or dusk, point2grid may not have any data for regridding (e.g, for high quality observed AOD ).  Therefore,  following warning message will appeared

(a) 
WARNING:  
WARNING: regrid_goes_variable() -> No valid data!
WARNING:  
DEBUG 2: Range of data (name="AOD"; level="(*,*)";)
DEBUG 2:        **input: -0.0138589 to 4.99997**    regridded: -9999 to -9999.

or
(b)
WARNING:  
WARNING: regrid_goes_variable() -> No valid data!
WARNING:  
DEBUG 2: Range of data (name="AOD"; level="(*,*)";)
DEBUG 2:        **input: -9999 to -9999**   regridded: -9999 to -9999.

For NCEP NCO operational requirement, can you turn WARNING to DEBUG for "regrid_goes_variable() -> No valid data!"?

[evs_aqmv708_prep_grid2grid_g16_aodC_high_20241016.log](https://github.com/user-attachments/files/17462577/evs_aqmv708_prep_grid2grid_g16_aodC_high_20241016.log)

Please tell me how do I send you the sample netcdf files.

Ho-Chun Huang

**Comment by Ho-ChunHuang-NOAA** [comment]
@malloryprow for your awareness

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @Ho-ChunHuang-NOAA, thanks for reaching out about this situation with the Point2Grid tool. We'd assumed that no valid output data would warrant a warning message being printed, but I understand that during overnight hours, its often the case that no observations actually meet the specified quality threshold.

I documented this with issue dtcenter/MET#3000 and fixed it with pull request dtcenter/MET#3001. In that PR, I changed 9 `WARNING` messages to `DEBUG(2)` log messages instead. I did retain some of the warning messages related to problems with processing the output grid, but those do still seem warranted.

After @hsoh-u reviewed my pull request, I merged in the changes, which will be included in the MET-12.0.0-rc1 release in about a month. The MET-12.0.0-beta6 already occurred at the end of last week. Hopefully, this is the resolution you wanted to see.

**Comment by Ho-ChunHuang-NOAA** [comment]
@JohnHalleyGotway Thank you for the quick response.  I will wait for the rc1 to be available on the WCOSS2to see the changes.  I can now continue the EVSv2.0 GOES AOD verification task.  Appreciated your effort.
